libs/asset-editor/src/lib/components/asset-editor-id-list/asset-editor-id-list.component.ts

```diff
@@ -11,85 +10,27 @@ import { IdVM } from '../../models';
 })
 export class AssetEditorIdListComponent {
   @Input({ required: true })
-  public control!: FormControl<IdVM[]>;
-
-  private _formValue: NewId | IdVM | null = null;
-
-  get ids(): IdVM[] {
-    return this.control.value;
-  }
-
-  @HostBinding('class.is-editing')
-  get isFormOpen(): boolean {
-    return this._formValue !== null || this.ids.length <= 1;
-  }
-
-  get formValue(): IdVM | null {
-    if (this._formValue === null) {
-      return this.ids.length === 1 ? this.ids[0] : null;
-    }
-    return this._formValue === NewId ? null : this._formValue;
-  }
-
-  get isNewFormOpen(): boolean {
-    return this._formValue === NewId;
-  }
-
-  isFormValue(id: IdVM): boolean {
-    return this._formValue === null || this._formValue === NewId
-      ? false
-      : O.toNullable(id.idId) === O.toNullable(this._formValue.idId);
-  }
-
-  openForm(id?: IdVM): void {
-    this._formValue = id ?? NewId;
-  }
-
-  saveId(id: IdVM): void {
-    const { formValue: oldId } = this;
-    if (oldId == null) {
-      this.control.setValue([...this.ids, id]);
-      return;
-    }
-    this.mutateId(oldId, (ids, i) => {
-      ids[i] = id;
-    });
-    this._formValue = id;
-  }
-
-  deleteId(id: IdVM): void {
-    this.mutateId(id, (ids, i) => {
-      ids.splice(i, 1);
+  public control!: FormControl<AlternativeId[]>;
+
+  public updateAlternativeId(value: string, field: keyof Omit<AlternativeId, 'idId'>, index: number): void {
```

Comment:
Being explicit is much easier to understand here:
```suggestion
  public updateAlternativeId(value: string, field: 'id' | 'description', index: number): void {
```